modules/cache2k/src/main/scala/scalacache/cache2k/Cache2kCache.scala

```diff
@@ -0,0 +1,62 @@
+package scalacache.cache2k
+
+import org.cache2k.{Cache => CCache}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration._
+import scala.language.higherKinds
+import scalacache.{AbstractCache, CacheConfig, Mode}
+
+/*
+ * Thin wrapper around cache2k.
+ *
+ * This cache implementation is synchronous.
+ */
+class Cache2kCache[V](underlying: CCache[String, V])(implicit val config: CacheConfig) extends AbstractCache[V] {
+
+  override protected final val logger =
+    LoggerFactory.getLogger(getClass.getName)
+
+  def doGet[F[_]](key: String)(implicit mode: Mode[F]): F[Option[V]] = {
+    mode.M.delay {
+      val result = Option(underlying.peek(key))
+      logCacheHitOrMiss(key, result)
+      result
+    }
+  }
+
+  def doPut[F[_]](key: String, value: V, ttl: Option[Duration])(implicit mode: Mode[F]): F[Any] = {
+    mode.M.delay {
+      underlying.put(key, value)
+      ttl.foreach(x => underlying.expireAt(key, toExpiryTime(x)))
+      logCachePut(key, ttl)
+    }
+  }
+
+  override def doRemove[F[_]](key: String)(implicit mode: Mode[F]): F[Any] =
+    mode.M.delay(underlying.remove(key))
+
+  override def doRemoveAll[F[_]]()(implicit mode: Mode[F]): F[Any] =
+    mode.M.delay(underlying.clear())
+
+  override def close[F[_]]()(implicit mode: Mode[F]): F[Any] = {
+    // Nothing to do
+    mode.M.pure(())
+  }
+
+  private def toExpiryTime(ttl: Duration): Long =
+    System.currentTimeMillis + ttl.toMillis
+
+}
+
+object Cache2kCache {
+
+  /**
+    * Create a new cache utilizing the given underlying cache2k cache.
+    *
+    * @param underlying a cache2k cache configured with a ExpiryPolicy or Cache2kBuilder.expireAfterWrite(long, TimeUnit)
```

Comment:
The given cache must be configured with a ExpiryPolicy or Cache2kBuilder.expireAfterWrite(long, TimeUnit)
https://cache2k.org/docs/1.0/apidocs/cache2k-api/org/cache2k/Cache.html#expireAt-K-long-